### DISCUSSION TITLE
open/close all buttons for accordion view

### DISCUSSION THREAD

**Discussion by esoterictriangle** [tabpanelwidget/tabpanelwidget#3] [Ideas]
Are there any thoughts on open/close buttons for the accordion view? In its most basic form could this be as simple as a function that adds `tpw-selected` to all headings when the open button is clicked & removes it from all when the close button is clicked?

**Comment by thierryk** [comment]
I like the idea to open all panels at once very much; mostly because it facilitates *searching* the content of the widget. 

We discussed including this option but we didn't prioritize it so it never shipped.

I'm not sure about the best way to do it though, nor if it is that easy to add.
There are styling considerations attached to the button (its placement, its look, etc.) as well as JS options for the behavior. For example, instead of applying a class to toggle the panels we could simply *unmount the widget*. This may be the simplest approach but the concern then would be related to styling (again) and accessibility (screen-reader users may not be able to get the markup change).

Also, should we create and expose the button for dynamic widgets (tabpanel that becomes an accordion) or only for static ones (widgets meant to be accordions only)?

Thanks for starting the conversation. Once we have a good picture of what's involved and what's best for authors and users we can work on something.
